hub/core/serialize.py

```diff
@@ -65,38 +65,34 @@ def serialize_chunk(
 
     # Write shape info
     if shape_info.ndim == 1:
-        flatbuff[offset : offset + 8] = np.zeros(8, dtype=np.byte)
+        flatbuff[offset : offset + 8] = np.zeros(8, dtype=np.byte).tobytes()
         offset += 8
     else:
-        flatbuff[offset : offset + 8] = np.array(shape_info.shape, dtype=np.int32).view(
-            np.byte
-        )
+        flatbuff[offset : offset + 8] = np.array(
+            shape_info.shape, dtype=np.int32
+        ).tobytes()
         offset += 8
-        flatbuff[offset : offset + shape_info.nbytes] = shape_info.reshape(-1).view(
-            np.byte
-        )
+        flatbuff[offset : offset + shape_info.nbytes] = shape_info.reshape(-1).tobytes()
         offset += shape_info.nbytes
 
     # Write byte positions
     if byte_positions.ndim == 1:
-        flatbuff[offset : offset + 4] = np.zeros(4, dtype=np.byte)
+        flatbuff[offset : offset + 4] = np.zeros(4, dtype=np.byte).tobytes()
         offset += 4
     else:
-        flatbuff[offset : offset + 4] = np.int32(byte_positions.shape[0]).view(
-            (np.byte, 4)
-        )
+        flatbuff[offset : offset + 4] = np.int32(byte_positions.shape[0]).tobytes()
         offset += 4
         flatbuff[offset : offset + byte_positions.nbytes] = byte_positions.reshape(
             -1
-        ).view(np.byte)
+        ).tobytes()
         offset += byte_positions.nbytes
 
     # Write actual data
     for byts in data:
         n = len(byts)
-        flatbuff[offset : offset + n] = np.frombuffer(byts, dtype=np.byte)
+        flatbuff[offset : offset + n] = byts
```

Comment:
this is the main part that leads to the speedup, earlier we were converting bytes to np array and then converting back to bytes after the for loop